### PR TITLE
Memory leak benchmarks (file and random integers streaming)

### PR DESCRIPTION
**Pull request by slvrtrn**
Related to #57

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,8 @@
 package-lock.json
 dist/
 .idea
-node_modules
\ No newline at end of file
+node_modules
+lib
```
Comment (mshustov) [inline .gitignore]: Maybe we should output into `build` folder?

File: benchmarks/leaks/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+```sh
+docker-compose up -d
+```
+
+#### Prepare input data
+
+```sh
+mkdir -p benchmarks/leaks/input \
+&& wget https://datasets.clickhouse.com/mgbench1.csv.xz \
```
Comment (mshustov) [inline benchmarks/leaks/README.md]: Why not curl? Not every person has `wget` install on the local env.
The same about `xz`.

File: benchmarks/leaks/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+```sh
+tsc --project benchmarks/tsconfig.benchmarks.json \
+&& node --expose-gc benchmarks/build/benchmarks/leaks/memory_leak_brown.js
+```
+
+
+## Random data streaming test
+
+---
```
Comment (mshustov) [inline benchmarks/leaks/README.md:49]: Does it mean `TODO`?

File: src/client.ts [hunk 4]
```diff
@@ -142,6 +142,7 @@ export class ClickHouseClient {
     return new Rows(stream, format)
   }
 
+  // FIXME: allow ; in the end of the query
```
Comment (mshustov) [inline src/client.ts:145]: What does it mean?

File: benchmarks/leaks/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+&& mv mgbench1.csv benchmarks/leaks/input \
+&& rm mgbench1.csv.xz
+```
+
+See [official examples](https://clickhouse.com/docs/en/getting-started/example-datasets/brown-benchmark/) for more information.
+
+#### Run the test
+
+```sh
+tsc --project benchmarks/tsconfig.benchmarks.json \
```
Comment (mshustov) [inline benchmarks/leaks/README.md]: Why don't you use `tsconfig.eslint.json` (`npm run build:all`)?
And we need to rename `tsconfig.eslint.json`, for sure.

File: benchmarks/leaks/shared.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+export interface MemoryUsage {
+  rss: number
+  heapTotal: number
+  heapUsed: number
+  external: number
+  arrayBuffers: number
+}
+
+export function getMemoryUsageInMegabytes(print = true): MemoryUsage {
```
Comment (mshustov) [inline benchmarks/leaks/shared.ts]: Why not separate print from formatting?

File: benchmarks/leaks/README.md (hunk 2 above)
Comment (slvrtrn) [reply benchmarks/leaks/README.md]: That's from the [docs](https://clickhouse.com/docs/en/getting-started/example-datasets/brown-benchmark/)

File: benchmarks/leaks/README.md (hunk 3 above)
Comment (slvrtrn) [reply benchmarks/leaks/README.md:49]: I committed the second part :)

File: src/client.ts (hunk 4 above)
Comment (slvrtrn) [reply src/client.ts:145]: we need to allow queries that end with a semicolon. Currently, `SELECT * FROM table;` will fail, logging that "multiline statements are not allowed", cause the client will try to attach `FORMAT` after the semicolon.

Comment (slvrtrn) [reply src/client.ts:145]: I will fix it in a follow-up PR

File: .gitignore (hunk 1 above)
Comment (slvrtrn) [reply .gitignore]: This is actually unused now, I will remove this from `gitignore`

File: benchmarks/leaks/README.md (hunk 5 above)
Comment (slvrtrn) [reply benchmarks/leaks/README.md]: A good one, I will try.

File: benchmarks/leaks/README.md (hunk 2 above)
Comment (slvrtrn) [reply benchmarks/leaks/README.md]: Unfortunately, `xz-utils` are required here. Could not do it with just `tar`

**Pull request merged**